modules/giapi/src/main/scala/giapi/client/GiapiConfig.scala

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2016-2018 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package giapi.client
+
+import cats.Show
+
+// Produce a configuration string for Giapi.
+trait GiapiConfig[T] {
+  def configuration(t: T): String
+}
+
+object GiapiConfig {
+  def apply[A](implicit instance: GiapiConfig[A]): GiapiConfig[A] = instance
+
+  implicit class GiapiConfigOps[A](value: A) {
```

Comment:
i think this can be a value class with `extends AnyVal`

Reply:
Maybe if we move this to `giapi.client` we can simplify the imports. Or perhaps create a `giapi.client.implicits` package containing it

Reply:
I would prefer to add this as a distinct syntax class following the pattern [here](https://github.com/gemini-hlsw/ocs3/tree/develop/modules/core/shared/src/main/scala/gem/syntax). It seems like a lot of boilerplate but I think it pays off to be consistent in how we handle this.

Reply:
Do you mean in a `giapi.client.syntax` package? Sounds good

Reply:
Yes, so `import giapi.client.syntax.giapiconfig._` or `import giapi.client.syntax.all._`.

Reply:
Okay. I'll make that change.